### PR TITLE
ARROW-7842: [Rust] [Parquet] Arrow list reader

### PR DESCRIPTION
**Pull request by nevi-me**
This is a port of #6770 to the parquet-writer branch.

We'll have more of a chance to test this reader,and ensure that we can roundtrip on list types.

### CODE REVIEW

File: rust/parquet/src/arrow/arrow_writer.rs [hunk 1]
```diff
@@ -1049,9 +1049,7 @@ mod tests {
     }
 
     #[test]
-    #[should_panic(
-        expected = "Reading parquet list array into arrow is not supported yet!"
-    )]
+    #[ignore = "Roundtrip failing, reason likely reader, not yet investigated"]
```
Comment (nevi-me) [inline rust/parquet/src/arrow/arrow_writer.rs:1052]: @carols10cents when you can, may you please help me out here on these 2 tests that I've ignored?
One needs a `LargeList` writer implementation, which I think you'd do better than me as you've done roundtrip for other Arrow types.

I've unignored `struct_single_column` as it's now passing, might be the null comparison commit that fixed it.

Thanks :smiley:

**Comment by nevi-me** [comment]
This doesn't need review, as I've addressed the questions that were in #6770. I'll wait for Carol's input before merging this.

CC @velvia @mcassels @maxburke as it was your original PR :)

File: rust/parquet/src/arrow/arrow_writer.rs (hunk 1 above)
Comment (carols10cents) [reply rust/parquet/src/arrow/arrow_writer.rs:1052]: Looking at this now!

Comment (carols10cents) [reply rust/parquet/src/arrow/arrow_writer.rs:1052]: @nevi-me I [opened a PR on your repo](https://github.com/nevi-me/arrow/pull/6) with some further work -- feel free to cherry pick or whatever, but the tests aren't quite passing yet over there, just failing for a different reason ;)

**Comment by nevi-me** [comment]
Merged

**Pull request closed without merging**